File: crossecore-java/src/java_/BooleanLiteralBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class BooleanLiteralBase 
extends ExpressionImpl implements BooleanLiteral
{
	protected static final boolean VALUE_EDEFAULT = false;
	private boolean _value = VALUE_EDEFAULT;
	public boolean isValue()
	{
		return _value;
	}
	public void setValue(boolean value){
		
		boolean oldValue = _value;
		_value = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.BOOLEAN_LITERAL__VALUE, oldValue, value));
		
	}
	
	
	

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.BOOLEANLITERAL;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.BOOLEAN_LITERAL__VALUE:
				return isValue();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.BOOLEAN_LITERAL__VALUE:
				setValue((boolean) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.BOOLEAN_LITERAL__VALUE:
				return _value != VALUE_EDEFAULT;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/BooleanLiteralBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class BooleanLiteralBase 
	:ExpressionImpl, BooleanLiteral
	{
		private bool _value = false;
		public virtual bool value
		{
		get { 
			return _value;
		}
		set { _value = value; }
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.BOOLEANLITERAL;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.BOOLEANLITERAL_VALUE:
					return value;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.BOOLEANLITERAL_VALUE:
					value = (bool) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.BOOLEANLITERAL_VALUE:
					return _value != VALUE_EDEFAULT;
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
